packages/frontend/styles/ddsIcons.ts

```diff
@@ -0,0 +1,472 @@
+    'filter-remove',
+    'folder-arrow-up',
+    'folder-media',
+    'gear-arrow',
+    'gear-wrench',
+    'graphics-card',
+    'hand-touch-screen',
+    'hd-generic',
+    'headphones',
+    'headset',
+    'heat',
+    'help-sqr',
+    'host-server',
+    'host-server-hex',
+    'host-server-rack',
+    'inbox-arrow-down',
+    'infrastructure',
+    'keypad',
+    'lcd',
+    'location-play',
+    'log-out',
+    'leaf',
+    'manage-info',
+    'management-software',
+    'memory',
+    'memory-card-reader',
+    'memory-sd-card',
+    'microchip',
+    'microphone-on',
+    'migrate',
+    'minimize',
+    '2-in-1',
+    'pre-seed',
+    'receipt',
+    'brand-xps',
+    'data-protection',
+    'hyper-converged-infrastructure',
+    'unpin',
+    'network-connected',
+    'network-horiz',
+    'network-share',
+    'network-vert-disable',
+    'partnership',
+    'password',
+    'pause-sqr',
+    'pencil-remove',
+    'popout-double',
+    'popout-share',
+    'popout-window-arrow',
+    'pop-up-arrow-corner',
+    'port-double',
+    'power-off',
+    'power-on',
+    'protractor',
+    'pulse',
+    'puzzle',
+    'qos',
+    'rack',
+    'rss',
+    'search-arrow-right',
+    'server-rack-arrow-right',
+    'shipping',
+    'skip-forward',
+    'skip-to-end',
+    'skip-to-front',
+    'slider-horiz',
+    'smart-select',
+    'social-networking',
+    'start-up',
+    'suspend',
+    'sync-vert',
+    'task-check',
+    'task-check-progress',
+    'usb',
+    'user-about',
+    'user-add',
+    'user-check',
+    'user-cir',
+    'user-cir-add',
+    'user-group-disable',
+    'user-groups-disable',
+    'user-lock',
+    'user-remove',
+    'user-search',
+    'user-unlock',
+    'video-card',
+    'view-detailed',
+    'web-camera',
+    'web-camera-hd',
+    'wifi-simple',
+    'windows-support',
+    'wrench-cir',
+    'wrench-double',
+    'package-closed',
+    'package-open',
+    'pie-chart',
+    'pie-chart-solid',
+    'power-supply-cord',
+    'power-supply-double',
+    'processor',
+    'productivity',
+    'removable-storage-usb',
+    'server-rack-search',
+    'server-rack-support',
+    'shield-protect',
+    'sort-za',
+    'speaker-double',
+    'stack',
+    'toolbox',
+    'touch-pad',
+    'translate',
+    'user-cir-remove',
+    'user-contact',
+    'user-disable',
+    'user-group-chat',
+    'view-list',
+    'view-row',
+    'view-table',
+    'virtualization',
+    'wrench-manage',
+    'zoom-in',
+    'zoom-out'
+];
+
+const iconClassName = (iconSuffix: string) => `dds__icon dds__icon--${iconSuffix}`;
+
+const dashToCamelCase = (text: string) => text.replace(/-([a-z])/g, dashPrefix => dashPrefix[1].toUpperCase());
+
+export const icons: Record<string, string> = iconsSuffix.reduce(
+    (iconsObj, iconSuffix) => ({
+        ...iconsObj,
+        [dashToCamelCase(iconSuffix)]: iconClassName(iconSuffix)
+    }),
+    {}
+);
+
+export const iconButtonClassName = (ddsIcon: string) => `icon-button ${ddsIcon}`
```

Comment:
WDYT About the following modification?

```suggestion
export const iconNames = [
'accessibility',
'addSqr',
'airplane',
'alarmBell',
'alarmClock',
'alertCheckCir',
'alertCheckSqr',
'alertError',
'alertInfoCir',
'alertNotice',
'appWindow',
'arrowCirDown',
'arrowCirLeft',
'arrowCirRight',
'arrowCirUp',
'arrowDown',
'arrowExport',
'arrowLeft',
'arrowRight',
'arrowTriDown',
'arrowTriLeft',
'arrowTriRight',
'arrowTriSolidRight',
'arrowTriUp',
'arrowUp',
'assistance',
'audioSpeaker',
'audioSpeakerMute',
'audioSpeakerOn',
'awardCertificate',
'awardTrophy',
'batteryCharge',
'bolt',
'bookmark',
'bookOpen',
'brightness',
'calculator',
'calendar',
'camera',
'cancelled',
'captureImage',
'cardCredit',
'cardInfo',
'carryCase',
'cart',
'chainLink',
'chargeBack',
'chartBarsAxis',
'chartBarsBox',
'chartInsights',
'chatBot',
'chatDots',
'check',
'chevronCirDown',
'chevronCirLeft',
'chevronCirRight',
'chevronCirUp',
'chevronDown',
'chevronLeft',
'chevronRight',
'chevronUp',
'chipSet',
'clipboard',
'clock',
'clockArrowLeft',
'clockArrowRight',
'clone',
'closeX',
'closeBack',
'closeCir',
'closeSqr',
'cloud',
'cloudCheck',
'collapseDownSqr',
'collapseUpSqr',
'comment',
'compare',
'conversation',
'conversationCir',
'copy',
'copyAlt',
'cube',
'currency',
'cursor',
'dDisplay',
'dashboard',
'dataCenter',
'dataCompute',
'datastore',
'debug',
'deviceDataCenter',
'deviceLaptop',
'deviceMobile',
'deviceModular',
'deviceMonitor',
'deviceServer',
'deviceServerTower',
'diagnostic',
'dimensionsWeight',
'discSoftware',
'docAdd',
'docArrowLeft',
'docArrowRight',
'docLines',
'docReports',
'document',
'downloadAlt',
'ear',
'energyEfficient',
'escalateTri',
'exitDoor',
'eyeViewOn',
'fan',
'filter',
'fingerprint',
'flag',
'flame',
'flashDrive',
'folder',
'folderAdd',
'folderArrowRight',
'fullScreen',
'gear',
'globeSimple',
'handle',
'handTouch',
'hardDrive',
'heart',
'heartFilled',
'helpCir',
'hinge',
'home',
'import',
'importAlt',
'inbox',
'inkToner',
'key',
'keyboard',
'keyboardMouse',
'laptopGeneric',
'lightbulb',
'location',
'lockClosed',
'lockOpen',
'logIn',
'mail',
'mailOpen',
'menuClosed',
'microphoneOff',
'minusMinimize',
'missingImage',
'mouse',
'mouseCord',
'movieClapper',
'movieClapperPlay',
'musicNote',
'networkVert',
'office',
'overflow',
'paintBrush',
'paintPalette',
'paperClip',
'partners',
'pause',
'pencil',
'pencilCheck',
'pencilMark',
'performance',
'phone',
'pin',
'playCir',
'popoutWindow',
'popUpArrow',
'powerSupply',
'printer',
'projector',
'recycle',
'recycleBin',
'redo',
'redoMove',
'refreshSpin',
'removeCir',
'removeSqr',
'reset',
'restore',
'save',
'saveDisk',
'scale',
'search',
'searchArrowLeft',
'searchBack',
'searchForward',
'share',
'shield',
'shieldCheck',
'sliderVert',
'snapshot',
'snowflake',
'sortHoriz',
'sortVert',
'star',
'starFilled',
'stop',
'syncHoriz',
'tag',
'thumbDown',
'thumbUp',
'trash',
'tv',
'undo',
'undoMove',
'upload',
'user',
'userGroup',
'userGroups',
'viewColumn',
'viewGrid',
'viewGridAlt',
'warranty',
'weightBell',
'wifi',
'wrenchSingle',
'wrenchTools',
'wifiNone',
'globeLocation',
'brandSkype',
'giftCard',
'metadata',
'objects',
'playSqr',
'plusAdd',
'sortAz',
'ellipsis',
'arrange',
'volume',
'arrowTriSolidStacked',
'popoutDoubleCondenseVertical',
'popoutDoubleCondense',
'barbell',
'expand',
'currencyCoinsPounds',
'currencyCoinsRupees',
'currencyCoinsYen',
'audioCard',
'popoutDoubleVertical',
'audioSpeakerLevels',
'deviceDesktop',
'audioSpeakerOff',
'calendarAdd',
'calendarCheck',
'calendarRemove',
'heartHalfFilled',
'batteryEmpty',
'starHalfFilled',
'thumbDownFilled',
'thumbUpFilled',
'brandAlienware',
'brandDell',
'brandDellemc',
'brandDelltechnologies',
'brandDelltechnologiesvert',
'bookmarkFilled',
'cardSwipeLeft',
'cardSwipeRight',
'chargeBackFolder',
'chargeBackReport',
'clipboardCheck',
'cloudDocument',
'cloudIq',
'columnAdd',
'commentAdd',
'commentCheck',
'deploy',
'deviceLaptopDetachable',
'deviceLockScreen',
'deviceStorageArray',
'deviceSwitchBlade',
'deviceSwitchBlade-2U',
'deviceTabletHoriz',
'deviceTabletVert',
'deviceThinClient',
'discArrowDown',
'discArrowUp',
'brandDellGaming',
'tagDeal',
'alertInfoSqr',
'userDirectory',
'brandWindows',
'appWindowSearch',
'currencyCoinsIndianRupees',
'currencyCoinsGeneric',
'currencyCoinsEuros',
'addCir',
'gameController',
'gamingDesktop',
'batteryHorizEmpty',
'batteryHorizFull',
'batteryHorizHalf',
'batteryVertFull',
'batteryVertHalf',
'bluetooth',
'calendarSearch',
'checkProperty',
'clipboardLines',
'clipboardNotes',
'clipboardRemove',
'clipboardSearch',
'cloudRestore',
'cloudUpload',
'cloudWireless',
'collapseDown',
'collapseUp',
'currencyCoins',
'cursorClock',
'deviceNodes',
'deviceScreenSize',
'discCheck',
'discInsert',
'discRecovery',
'discSecure',
'discSystem',
'docCheck',
'docCube',
'docManual',
'docsCopyAlt',
'docSearch',
'docSecure',
'docSettings',
'docSupport',
'download',
'eyeViewOff',
'filterAdd',
'filterClear',
'filterMove',
'filterRemove',
'folderArrowUp',
'folderMedia',
'gearArrow',
'gearWrench',
'graphicsCard',
'handTouchScreen',
'hdGeneric',
'headphones',
'headset',
'heat',
'helpSqr',
'hostServer',
'hostServerHex',
'hostServerRack',
'inboxArrowDown',
'infrastructure',
'keypad',
'lcd',
'locationPlay',
'logOut',
'leaf',
'manageInfo',
'managementSoftware',
'memory',
'memoryCardReader',
'memorySdCard',
'microchip',
'microphoneOn',
'migrate',
'minimize',
'2In-1',
'preSeed',
'receipt',
'brandXps',
'dataProtection',
'hyperConvergedInfrastructure',
'unpin',
'networkConnected',
'networkHoriz',
'networkShare',
'networkVertDisable',
'partnership',
'password',
'pauseSqr',
'pencilRemove',
'popoutDouble',
'popoutShare',
'popoutWindowArrow',
'popUpArrowCorner',
'portDouble',
'powerOff',
'powerOn',
'protractor',
'pulse',
'puzzle',
'qos',
'rack',
'rss',
'searchArrowRight',
'serverRackArrowRight',
'shipping',
'skipForward',
'skipToEnd',
'skipToFront',
'sliderHoriz',
'smartSelect',
'socialNetworking',
'startUp',
'suspend',
'syncVert',
'taskCheck',
'taskCheckProgress',
'usb',
'userAbout',
'userAdd',
'userCheck',
'userCir',
'userCirAdd',
'userGroupDisable',
'userGroupsDisable',
'userLock',
'userRemove',
'userSearch',
'userUnlock',
'videoCard',
'viewDetailed',
'webCamera',
'webCameraHd',
'wifiSimple',
'windowsSupport',
'wrenchCir',
'wrenchDouble',
'packageClosed',
'packageOpen',
'pieChart',
'pieChartSolid',
'powerSupplyCord',
'powerSupplyDouble',
'processor',
'productivity',
'removableStorageUsb',
'serverRackSearch',
'serverRackSupport',
'shieldProtect',
'sortZa',
'speakerDouble',
'stack',
'toolbox',
'touchPad',
'translate',
'userCirRemove',
'userContact',
'userDisable',
'userGroupChat',
'viewList',
'viewRow',
'viewTable',
'virtualization',
'wrenchManage',
'zoomIn',
'zoomOut'
] as const;

const iconNameToIconClassName = (iconName: string) => iconName.replace(/[A-Z]/g, upperCaseLetter => `-${upperCaseLetter.toLowerCase()}`);

const iconClassName = (iconName: string) => {
    const iconClassName = iconNameToIconClassName(iconName);
    return `dds__icon dds__icon--${iconClassName}`;
};

export const iconButtonClassName = (iconName: string) => {
    const iconClassName = iconNameToIconClassName(iconName);
    return `icon-button ${iconClassName}`;
}

type Icons = Record<typeof iconNames[number], string>;

export const icons: Icons = iconNames.reduce(
    (iconsObj, iconName) => ({
        ...iconsObj,
        [iconName]: iconClassName(iconName)
    }),
    {} as Icons
);

```

It's modifying the idea behind this code by 180 degrees, but it's giving us the ability to have  a typing suggestions regarding icon names
In addition, we could omit typing icon suffixes by hand, in favor of having a dedicated object, which we could use to implement desired icon 🎉 

With previous approach, we'd have to do 
```typescript
<i className={icons['audio-speaker-mute']} />
```

But currently we should be able to do
```typescript
<i className={icons[iconNames.audioSpeakerMute]} />
```

which should be less error prone and more friendly to developers 🎉 

Additional benefit is that we are having a typescript support under the hood, so making
```typescript
<i className={icons['some-random-text']} />
```
will result in getting an error - which wasn't the case with the previous approach

I know that approach modification comes with the great amount of modified lines of code
That's why I thought that I'll make a ready to implement suggestion, as I can see a lot of added value coming out of the presented approach 💪 

Let me know what do you think about it 😉

Reply:
[ec72503](https://github.com/cloudify-cosmo/cloudify-ui-common/pull/164/commits/ec72503cfec39642f0602fec8147986a7f14235a)